student-search.go

```diff
@@ -5,83 +5,73 @@ import (
 
 	"github.com/pclubiitk/student-search/database"
 
-	"github.com/iris-contrib/middleware/cors"
-	"github.com/iris-contrib/middleware/logger"
-	"github.com/iris-contrib/middleware/recovery"
+	"github.com/go-pg/pg"
+	"github.com/labstack/echo"
+	"github.com/labstack/echo/middleware"
 	"github.com/olebedev/config"
-	"gopkg.in/kataras/iris.v5"
-	"gopkg.in/pg.v5"
 )
 
-func main() {
+type serverConfig struct {
+	pghost   string
+	pgport   string
+	username string
+	dbname   string
+	httpport int
+}
 
-	iris.Config.Gzip = true
-	iris.Config.LoggerPreffix = "[student-search] "
+func main() {
 
-	iris.Use(logger.New())
-	iris.Use(recovery.New())
-	iris.Use(cors.Default())
+	e := echo.New()
+	e.Use(middleware.Gzip())
```

Comment:
From the docs, it seems this could be simplified to:
```
loggerConf := middleware.LoggerConfig{
    Format: "[student-search] ${remote_ip} - - [${time_rfc3339}] \"${method} ${uri}\" ${status} ${bytes_out}\n",
}
e.Use(middleware.Gzip(), middleware.Recover(), middleware.LoggerWithConfig(loggerConf))
```